tips/TIP-0046/tip-0046.md

```diff
@@ -0,0 +1,961 @@
+
+Block (json-encoded):
+
+```json
+{
+  "protocolVersion": 3,
+  "networkId": "0",
+  "issuingTime": "1695275834000000000",
+  "slotCommitmentId": "0x8633b2eb1845fdecf12ee6c5e789c3cf1f0d0bbb3cee65cb5fb2757e995b5cd700000000",
+  "latestFinalizedSlot": 0,
+  "issuerId": "0x554813020e6324372e4b2018284c1403321457351a620a227a2e68201d325471",
+  "block": {
+    "type": 0,
+    "strongParents": [
+      "0x580b62530a1a4f2f6a25442448043a0d324c7579291a5c144742314c3a481e667035040e",
+      "0x5a29685d615d331e38684c0d1c7a2b7875243c77322048215c5c6740562c146155747479",
+      "0x7c602d30626b1056351c464b03672f260c50730f344b1152491451445e18653f3a515567"
+    ],
+    "weakParents": [],
+    "shallowLikeParents": [],
+    "payload": {
+      "type": 1,
+      "transaction": {
+        "networkId": "14147312347886322761",
+        "creationSlot": 0,
+        "contextInputs": [],
+        "inputs": [
+          {
+            "type": 0,
+            "transactionId": "0x3b27570e163e3c283e744d305a57473c51624e3e4c093865592576054025411920390b7c",
+            "transactionOutputIndex": 57
+          },
+          {
+            "type": 0,
+            "transactionId": "0x1c0c0c3e6740486b175a333c4a414b295036795b1e67045f6b405b790968750d6352770a",
+            "transactionOutputIndex": 114
+          }
+        ],
+        "allotments": [
+          {
+            "accountID": "0x470b7b276e6c644b7b3577213c7b3874183d677831596e667702386952432e68",
+            "value": "8343"
+          }
+        ],
+        "capabilities": "",
+        "outputs": [
+          {
+            "type": 0,
+            "amount": "5892",
+            "mana": "0",
+            "unlockConditions": [
+              {
+                "type": 0,
+                "address": {
+                  "type": 0,
+                  "pubKeyHash": "0x286a580f29774c5b7b7d0e6a7610303a277b124e6427363f1635161c753f3e53"
+                }
+              }
+            ],
+            "features": []
+          },
+          {
+            "type": 0,
+            "amount": "9003",
+            "mana": "0",
+            "unlockConditions": [
+              {
+                "type": 0,
+                "address": {
+                  "type": 0,
+                  "pubKeyHash": "0x7a0d271a7b347e7e614c5463683b405b357a2b0c594a22340b6a1d1a2d667a26"
+                }
+              }
+            ],
+            "features": []
+          }
+        ]
+      },
+      "unlocks": [
+        {
+          "type": 0,
+          "signature": {
+            "type": 0,
+            "publicKey": "0x0d2128337477155d5e48225e6c500b2538124611456c7739447031592f1d4d1d",
+            "signature": "0x55414109320a16182c162c1022713126396c4330505352550608266d3c5a364274525914514475730030295e0e0b1a7e6b367a236348382a6e625147512b3813"
+          }
+        },
+        {
+          "type": 0,
+          "signature": {
+            "type": 0,
+            "publicKey": "0x4459093a134a3b285d3c720c4d5c415204700d0c5c09525b340905233e1a7d61",
```

Comment:
this isn't a valid public key

Reply:
In what way is it not a valid public key?

Reply:
Not all 32 random bytes are valid Ed25519 public keys. In https://zips.z.cash/protocol/protocol.pdf#concreteed25519 you will see that the bytes be an encoding of a point `A` on the twisted Edwards form of Curve25519. Most crypto libraries, like the one we are using, are validating these bytes at construction.

Reply:
I see. My bad for using random data. Updated the test vectors with properly signed transactions and blocks.

Reply:
No worries, thanks a lot 🙏🏻